templates/aws-stack.yml

```diff
@@ -739,6 +750,36 @@ Resources:
       PolicyName: InstancePolicy
       PolicyDocument:
         Statement:
+          - !If
+            - HasSecretsBucket
+            - Effect: Allow
+              Action:
+                - s3:Get*
+                - s3:List*
+              Resource:
+                - !Sub
+                  - "arn:aws:s3:::${Bucket}/*"
+                  - Bucket: !If [ CreateSecretsBucket, !Ref ManagedSecretsBucket, !Ref SecretsBucket ]
+                - !Sub
+                  - "arn:aws:s3:::${Bucket}"
+                  - Bucket: !If [ CreateSecretsBucket, !Ref ManagedSecretsBucket, !Ref SecretsBucket ]
+            - !Ref AWS::NoValue
+          - !If
+            - UseArtifactsBucket
+            - Effect: Allow
+              Action:
+                - s3:GetObject
+                - s3:GetObjectAcl
+                - s3:GetObjectVersion
+                - s3:GetObjectVersionAcl
+                - s3:ListBucket
+                - s3:PutObject
+                - s3:PutObjectAcl
+                - s3:PutObjectVersionAcl
+              Resource:
+                - !Sub "arn:aws:s3:::${ArtifactsBucket}/*"
+                - !Sub "arn:aws:s3:::${ArtifactsBucket}"
+            - !Ref AWS::NoValue
```

Comment:
A small part of me likes that these were separate policies, with names indicating what they're for.
But not a big part.
Maybe we could add descriptive statement IDs (`Sid`) to give them descriptive names at least?

Reply:
I agree, though there is no way to conditionally `DependsOn` them because they were themselves `Condition`d resources.

We could do the same trick as for the VPC with the policies, but I think it’s less to think about having them conditionally shimmed into the `IAMPolicies` resource that can be unconditionally `DependsOn`d.

Adding an `Sid` is a good idea :+1: our other option would be to make these managed policies which are included in the role directly creating a suitable dependency chain. `ASG -> LaunchTemplate -> InstanceProfile -> IAMRole ->> ManagedPolicies`